### PR TITLE
fix: trpc/react-query — only one query client should exist

### PR DESCRIPTION
**Pull request by hbjORbj**
## What does this PR do?

- Wrote thoroughly in https://app.campsite.com/cal/posts/l8yff3b6i8qq#comment-ddfo3lvovjuq

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A - I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

### CODE REVIEW

File: apps/web/app/_trpc/trpc-provider.tsx [hunk 1]
```diff
@@ -32,16 +37,36 @@ const resolveEndpoint = (links: any) => {
...
-export const TrpcProvider: React.FC<{ children: React.ReactNode; dehydratedState?: DehydratedState }> = ({
-  children,
-  dehydratedState,
-}) => {
+export const TrpcProvider: React.FC<{ children: React.ReactNode }> = ({ children }) => {
   const [queryClient] = useState(
     () =>
       new QueryClient({
-        defaultOptions: { queries: { staleTime: 5000 } },
+        // these configurations are copied from "packages/trpc/react/trpc.ts"
```
Comment (hbjORbj) [inline apps/web/app/_trpc/trpc-provider.tsx:44]: as the comment says

File: apps/web/app/_trpc/trpc-provider.tsx [hunk 2]
```diff
@@ -1,11 +1,16 @@
-import { type DehydratedState, QueryClient, QueryClientProvider } from "@tanstack/react-query";
-import { HydrateClient } from "app/_trpc/HydrateClient";
+"use client";
```
Comment (hbjORbj) [inline apps/web/app/_trpc/trpc-provider.tsx:1]: this is a client component. we use useState.

File: apps/web/app/layout.tsx [hunk 3]
```diff
@@ -112,7 +113,7 @@ export default async function RootLayout({ children }: { children: React.ReactNo
             }}
           />
         )}
-        {children}
+        <TrpcProvider>{children}</TrpcProvider>
```
Comment (hbjORbj) [inline apps/web/app/layout.tsx:116]: We need to call this provider only once and hence it needs to be in this file

**Review by emrysal — APPROVED** [review]
🚀  nice find!

**Pull request merged**